Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/AbilData.xml

```diff
@@ -2308,36 +2308,76 @@
-        <InfoArray index="Research12" Time="200" Upgrade="AP_TerranInfantryWeaponsUltraCapacitorsLevel2">
+        <InfoArray index="Research12" Time="190" Upgrade="AP_TerranInfantryWeaponsUltraCapacitorsLevel2">
             <Resource index="Minerals" value="175"/>
             <Resource index="Vespene" value="175"/>
             <Button DefaultButtonFace="AP_TerranInfantryWeaponsUltraCapacitorsLevel2" State="Restricted" Requirements="AP_LearnTerranInfantryWeaponUltraCapacitors2"/>
         </InfoArray>
-        <InfoArray index="Research13" Time="230" Upgrade="AP_TerranInfantryWeaponsUltraCapacitorsLevel3">
+        <InfoArray index="Research13" Time="220" Upgrade="AP_TerranInfantryWeaponsUltraCapacitorsLevel3">
             <Resource index="Minerals" value="250"/>
             <Resource index="Vespene" value="250"/>
             <Button DefaultButtonFace="AP_TerranInfantryWeaponsUltraCapacitorsLevel3" State="Restricted" Requirements="AP_LearnTerranInfantryWeaponUltraCapacitors3"/>
         </InfoArray>
-        <InfoArray index="Research14" Time="170" Upgrade="AP_TerranInfantryArmorsVanadiumPlatingLevel1">
+        <InfoArray index="Research14" Time="160" Upgrade="AP_TerranInfantryArmorsVanadiumPlatingLevel1">
             <Resource index="Minerals" value="100"/>
             <Resource index="Vespene" value="100"/>
             <Button DefaultButtonFace="AP_TerranInfantryArmorVanadiumPlatingLevel1" State="Restricted" Requirements="AP_LearnTerranInfantryArmorVanadiumPlating1"/>
         </InfoArray>
-        <InfoArray index="Research15" Time="200" Upgrade="AP_TerranInfantryArmorsVanadiumPlatingLevel2">
+        <InfoArray index="Research15" Time="190" Upgrade="AP_TerranInfantryArmorsVanadiumPlatingLevel2">
             <Resource index="Minerals" value="175"/>
             <Resource index="Vespene" value="175"/>
             <Button DefaultButtonFace="AP_TerranInfantryArmorVanadiumPlatingLevel2" State="Restricted" Requirements="AP_LearnTerranInfantryArmorVanadiumPlating2"/>
         </InfoArray>
-        <InfoArray index="Research16" Time="230" Upgrade="AP_TerranInfantryArmorsVanadiumPlatingLevel3">
+        <InfoArray index="Research16" Time="220" Upgrade="AP_TerranInfantryArmorsVanadiumPlatingLevel3">
             <Resource index="Minerals" value="250"/>
             <Resource index="Vespene" value="250"/>
             <Button DefaultButtonFace="AP_TerranInfantryArmorVanadiumPlatingLevel3" State="Restricted" Requirements="AP_LearnTerranInfantryArmorVanadiumPlating3"/>
         </InfoArray>
+        <InfoArray index="Research17" Time="250" Upgrade="AP_TerranInfantryWeaponsLevel4">
+            <Resource index="Minerals" value="325"/>
+            <Resource index="Vespene" value="325"/>
+            <Button DefaultButtonFace="AP_TerranInfantryWeaponsLevel4" State="Restricted" Requirements="AP_LearnTerranInfantryWeapon4"/>
+        </InfoArray>
+        <InfoArray index="Research18" Time="280" Upgrade="AP_TerranInfantryWeaponsLevel5">

```

Comment:
I'd caution against setting the time to research too high, I'd even recommend capping the times at level 3 times. 160+190+220+250+280=1100 or about 18 minutes 20 seconds. That's longer than the vast majority of missions.

Checking co-op game data, Tychus upgrades are 120+140+160+180+200 = 800. That affects fewer units, so maybe we don't go that low. I suppose an alternative is to also provide research speed upgrades / have a way to give the ebay chrono boost somehow.

Reply:
Done by reducing all W/A research times if L4 or L5 is available so the total research time is capped at 570